operator/src/main/java/oracle/kubernetes/operator/utils/ValidationUtils.java

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.utils;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+
+import oracle.kubernetes.operator.logging.LoggingFacade;
+import oracle.kubernetes.operator.logging.LoggingFactory;
+import oracle.kubernetes.weblogic.domain.model.Cluster;
+import oracle.kubernetes.weblogic.domain.model.ClusterStatus;
+import oracle.kubernetes.weblogic.domain.model.Domain;
+import oracle.kubernetes.weblogic.domain.model.DomainSpec;
+import oracle.kubernetes.weblogic.domain.model.DomainStatus;
+import org.jetbrains.annotations.NotNull;
+
+import static org.apache.commons.collections4.CollectionUtils.isEqualCollection;
+
+/**
+ * ValidationUtil provides the utility methods to perform the validation of a proposed change to an existing
+ * Domain or Cluster resource. It is used by the validating webhook.
+ *
+ * <p>The current version supports validation of the following:
+ * <ul>
+ * <li>The proposed replicas settings at the domain level and/or cluster level can be honored by WebLogic domain config.
+ * </li>
+ * </ul>
+ * </p>
+ */
+
+public class ValidationUtils {
+  private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Webhook", "Operator");
+
+  private ValidationUtils() {
+  }
+
+  /**
+   * Validating a proposed domain resource against an existing domain resource. It returns true if the proposed changes
+   * in the proposed domain resource can be honored, otherwise, returns false.
+   *
+   * @param existingDomain domain that needs to be validated against.
+   * @param proposedDomain domain that needs to be validated.
+   * @return true if valid, otherwise false
+   */
+  public static boolean validateDomain(@NotNull Domain existingDomain, @NotNull Domain proposedDomain) {
```

Comment:
The name is a bit confusing: it's not actually "validating" a domain, is it? I infer that it is actually returning true if the proposed change is valid. If so, it would probably be better to call it something like, `isPermittedChange` or `isDomainReplacementAllowed`. Methods that return booleans are conventionally given names that are predicates.